src/core/ruby_version.rs

```diff
@@ -1749,56 +2030,64 @@ mod tests {
     fn real_stack_trace_3_3_0() -> Vec<StackFrame> {
         vec![
             StackFrame {
-                name: "sleep [c function]".to_string(),
+                name: "Kernel#sleep [c function]".to_string(),
                 relative_path: "(unknown)".to_string(),
                 absolute_path: None,
                 lineno: None,
             },
             StackFrame {
                 name: "Object#aaa".to_string(),
-                relative_path: "ci/ruby-programs/infinite.rb".to_string(),
+                relative_path: "ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 absolute_path: Some(
-                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite.rb".to_string(),
+                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 ),
-                lineno: Some(3),
+                lineno: Some(9),
             },
             StackFrame {
                 name: "Object#bbb".to_string(),
-                relative_path: "ci/ruby-programs/infinite.rb".to_string(),
+                relative_path: "ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 absolute_path: Some(
-                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite.rb".to_string(),
+                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 ),
-                lineno: Some(7),
+                lineno: Some(13),
             },
             StackFrame {
                 name: "Object#ccc".to_string(),
-                relative_path: "ci/ruby-programs/infinite.rb".to_string(),
+                relative_path: "ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 absolute_path: Some(
-                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite.rb".to_string(),
+                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 ),
-                lineno: Some(11),
+                lineno: Some(17),
             },
             StackFrame {
                 name: "block in <main>".to_string(),
-                relative_path: "ci/ruby-programs/infinite.rb".to_string(),
+                relative_path: "ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 absolute_path: Some(
-                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite.rb".to_string(),
+                    "/home/runner/work/rbspy/rbspy/ci/ruby-programs/infinite_on_cpu.rb".to_string(),
                 ),
-                lineno: Some(15),
+                lineno: Some(21),
             },
             StackFrame {
-                name: "loop".to_string(),
+                name: "Kernel#loop".to_string(),
                 relative_path: "<internal:kernel>".to_string(),
                 absolute_path: Some("unknown".to_string()),
                 lineno: Some(192),
             },
+            StackFrame {
```

Comment:
The coredump was invalid and this frame was dropped entirely. I redid the coredump and added the missing frame.